pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository.
+	JenkinsfilePath string `json:"jenkinsfilePath,omitempty"`
+
+	// Jenkinsfile is the raw contents of a Jenkinsfile containing the pipeline definition.
+	Jenkinsfile string `json:"jenkinsfile,omitempty"`
+
+	// Env contains additional environment variables you want to pass to the Jenkins Pipeline.
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
Well we haven't figured out how to yet (haven't spent time tbh) but we feel it's certainly something users will need. We currently load env vars from a known file in source repo - we'd like to be able to merge that env var file with env vars specified in BC.

@rawlingsj Any thoughts on this?